### PR TITLE
feat: Context menu on thread

### PR DESCRIPTION
**Pull request by lebojo**
Context menu when email selected act like right click on macos

### CODE REVIEW

**Review by lebojo — COMMENTED** [review]
Updated Loco

File: MailCore/Cache/Actions/ActionsManager.swift [hunk 1]
```diff
@@ -217,6 +217,9 @@ public class ActionsManager: ObservableObject {
             Task { @MainActor in
                 origin.nearestShareMailLinkPanel?.wrappedValue = result
             }
+        case .activeMultiselect:
+            Task { @MainActor in
+            }
```
[the remark this answers is not in the document]
Comment (lebojo) [reply MailCore/Cache/Actions/ActionsManager.swift]: Removed

File: Mail/Views/Thread List/ThreadListCellContextMenu.swift [hunk 2]
```diff
@@ -23,39 +23,48 @@ import SwiftModalPresentation
...
-                        Task {
-                            try await actionsManager.performAction(
-                                target: thread.messages.toArray(),
-                                action: action,
-                                origin: .swipe(originFolder: thread.folder, nearestMessagesToMoveSheet: $messagesToMove)
-                            )
+                    Button(role: isDestructiveAction(action)) {
+                        if action == .activeMultiselect {
+                            toggleMultipleSelection(false)
+                        } else {
```
Comment (PhilippeWeidmann) [inline Mail/Views/Thread List/ThreadListCellContextMenu.swift:51]: Use guard

File: MailCore/Cache/Actions/Action+List.swift [hunk 3]
```diff
@@ -387,4 +388,10 @@ public extension Action {
         iconResource: MailResourcesAsset.bin,
         matomoName: "deleteAccount"
     )
+    static let activeMultiselect = Action(
```
Comment (PhilippeWeidmann) [inline MailCore/Cache/Actions/Action+List.swift:391]: `enableMultipleSelection`

File: Mail/Views/Thread List/ThreadListCellContextMenu.swift [hunk 4]
```diff
@@ -64,4 +73,11 @@ struct ThreadListCellContextMenu: ViewModifier {
                     .sheetViewStyle()
             }
     }
+
+    private func isDestructiveAction(_ action: Action) -> ButtonRole? {
+        guard action == .archive else {
```
Comment (PhilippeWeidmann) [inline Mail/Views/Thread List/ThreadListCellContextMenu.swift:78]: Inverse guard

**Review by PhilippeWeidmann — APPROVED** [review]

File: MailCore/Cache/Actions/Action+List.swift [hunk 5]
```diff
@@ -86,7 +87,7 @@ extension Action: CaseIterable {
         let star = message.flagged
         let spam = message.folder?.role == .spam
         let print = origin.type == .floatingPanel(source: .messageList)
-        let tempListActions: [Action?] = [
+        var tempListActions: [Action?] = [
```
Comment (valentinperignon) [inline MailCore/Cache/Actions/Action+List.swift:90]: `temListActions` can still be a `let`

**Pull request merged**